docs/user-guide/self_rewarding.rst

```diff
@@ -0,0 +1,214 @@
+.. include:: /content/nemo.rsts
+
+Model Alignment by Self-Rewarding Language Models
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+Original paper: https://arxiv.org/abs/2401.10020
+Meta Self-Rewarding paper: https://arxiv.org/abs/2407.19594
+
+The NeMo framework supports efficient model alignment via the NeMo Aligner codebase.
+
+All algorithms in NeMo Aligner will work with any GPT based model that is from mcore(i.e in the config it has ``mcore_gpt=True``). For the purposes of this tutorial, we will go through the entire Self-Rewarding pipeline using the newly released `2B GPT model with 4096 sequence length <https://huggingface.co/nvidia/GPT-2B-001>`__.  This same tutorial also works for other GPT models(such as LLaMa2) of any size.
```

Comment:
Reorder text (move references to end of section), spell out Megatron Core, fix punctuation, fix capitalization, revise links. Suggested revision:

The NeMo Framework supports efficient model alignment using the NeMo Aligner codebase.

All algorithms in NeMo Aligner are compatible with any GPT-based model from Megatron Core (i.e., those with ``mcore_gpt=True`` in the configuration). For this tutorial, we will demonstrate the entire self-rewarding pipeline using the newly released `2B GPT model with 4096 sequence length <https://huggingface.co/nvidia/GPT-2B-001>`__. This tutorial is also applicable to other GPT models, such as LLaMa2, regardless of their size.

For more information, see the original `Self-Rewarding Language Models paper <https://arxiv.org/abs/2401.10020>`__ and the `Meta-Rewarding Language Models paper <https://arxiv.org/abs/2407.19594>`__.